packages/cryptography/src/Encryption/EncryptionAlgorithm.php

```diff
@@ -0,0 +1,30 @@
+<?php
+
+namespace Tempest\Cryptography\Encryption;
+
+enum EncryptionAlgorithm: string
+{
+    case AES_256_GCM = 'aes-256-gcm';
+    case AES_256_CBC = 'aes-256-cbc';
+    case AES_128_GCM = 'aes-128-gcm';
+    case AES_128_CBC = 'aes-128-cbc';
+    case CHACHA20_POLY1305 = 'chacha20-poly1305';
+
+    public function getKeyLength(): int
+    {
+        return openssl_cipher_key_length($this->value);
+    }
+
+    public function getIvLength(): int
+    {
+        return openssl_cipher_iv_length($this->value);
+    }
+
+    public function isAead(): bool
```

Comment:
Can we document what `isAhead` means exactly? I assume it's to signify some kind of encryption strenght?

Reply:
Yeah sure, I'll add a comment. `AE` is "authenticated encryption" means that it's encrypted and cannot be tampered with, but the important part is `AD` which means "associated data", which means that the message may contain de-encodable but non-tamperable arbitrary data